src/C4HudBars.h

```diff
@@ -0,0 +1,203 @@
+/*
+ * LegacyClonk
+ *
+ * Copyright (c) 2021, The LegacyClonk Team and contributors
+ *
+ * Distributed under the terms of the ISC license; see accompanying file
+ * "COPYING" for details.
+ *
+ * "Clonk" is a registered trademark of Matthes Bender, used with permission.
+ * See accompanying file "TRADEMARK" for details.
+ *
+ * To redistribute this file separately, substitute the full license texts
+ * for the above references.
+ */
+
+/* Allows to define custom energy bars */
+
+#pragma once
+
+#include "C4FacetEx.h"
+#include "C4Id.h"
+#include "C4Value.h"
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+struct C4AulContext;
+class C4Object;
+
+class C4HudBarDef
+{
+public:
+	enum Physical {
+		EBP_None   = 0,
+		EBP_Energy = 1,
+		EBP_Magic  = 2,
+		EBP_Breath = 3,
+		EBP_All = EBP_Energy | EBP_Magic | EBP_Breath
```

Comment:
EBP_All is maybe not a good name for this, it is used as a bitfield here to error if a non existent value is used in the definition:
https://github.com/legacyclonk/LegacyClonk/blob/8db83176cae5f1c9efca4defd069bcac46955d8c/src/C4HudBars.cpp#L572
So basically it is only used internally, maybe instead of having it as an enum member it should be an internal constant?

The idea behind or-ing all the enum values is, although it just boils down to 3, it better communicates the intent of having all the used bits in there, thinking about it this way EBP_None could even be added.
I don't feel strongly about any of this, so we can change how this works, as long as the valid values check still works.

Reply:
I think the more conventional way to do this is with a First and Last value and then checking if the value is in range.
This then also works correctly if the range of values is smaller than that of all involved bits.

Reply:
Simply use EBP_None as first and EBP_Breath as last, or would you define `EBP_First = EBP_None` and `EBP_Last = EBP_Breath` as explicit aliases?
I guess the test would then be `if(C4HudBarDef::EBP_First <= _physical && _physical <= C4HudBarDef::EBP_Last)`

Reply:
Just had another look at the CompileFunc, was worried it might extract names automatically, but it maps them manually...
It seems there is no change needed there, so I agree.
https://github.com/legacyclonk/LegacyClonk/blob/8db83176cae5f1c9efca4defd069bcac46955d8c/src/C4HudBars.cpp#L236-L242